code/game/objects/structures/headpike.dm

```diff
@@ -33,9 +33,9 @@
 	if(!victim) //likely a mapspawned one
 		victim = new(src)
 		victim.real_name = random_unique_name(prob(50))
-	spear = locate(speartype) in parts_list
+	spear = locate(/obj/item/spear) in parts_list
 	if(!spear)
-		spear = new speartype(src)
+		spear = new speartype(/obj/item/spear)
```

Comment:
revert

to clarify what this is doing
`thing = new something(at location)`
the arguments `( )` after `new` designates the location where the atom will be created.

Reply:
alright I had thought the spear only becomes null in this case if its maploaded like the abov

Reply:
> https://github.com/tgstation/tgstation/blob/79b0e91f6cc015af484069ff19cfadb7c3a90eb4/code/game/objects/structures/headpike.dm#L9
> 
> Change this to ` var/obj/item/spear/speartype = /obj/item/spear`
> 
> This will completely fix the problem without needing to alter any other code.

yeah thanks that solved the problem much better its commited and I reverted all of it like you said